wildfly-wasm/injection/src/main/java/org/wildfly/extension/wasm/injection/WasmPortableExtension.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.extension.wasm.injection;
+
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.AfterBeanDiscovery;
+import jakarta.enterprise.inject.spi.Extension;
+import java.util.Map;
+import org.wildfly.wasm.api.WasmInvoker;
+import org.wildfly.wasm.api.WasmTool.WasmToolLiteral;
+
+public class WasmPortableExtension implements Extension {
+
+    private final WildFlyWasmRegistry registry;
+
+    public WasmPortableExtension(WildFlyWasmRegistry registry) {
+        this.registry = registry;
+    }
+
+    public void atd(@Observes AfterBeanDiscovery atd) throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException {
```

Comment:
Nit: I figure this method and param name should be 'abd'. I assume atd means AfterTypeDiscovery.